### PR TITLE
Adding some more HTTP status constants

### PR DESCRIPTION
**Pull request by msva**
Some usefull status constants + prerequisite for #397

### CODE REVIEW

**Comment by Sirupsen** [comment]
Would like to see this in too! We use a few of those as well.

File: src/ngx_http_lua_consts.c [hunk 1]
```diff
@@ -89,14 +89,31 @@ ngx_http_lua_inject_http_consts(lua_State *L)
...
+    lua_pushinteger(L, NGX_HTTP_NO_CONTENT);
+    lua_setfield(L, -2, "HTTP_NO_CONTENT");
+
+    lua_pushinteger(L, NGX_HTTP_PARTIAL_CONTENT);
+    lua_setfield(L, -2, "HTTP_PARTIAL_CONTENT");
+
+
     lua_pushinteger(L, NGX_HTTP_SPECIAL_RESPONSE);
-    lua_setfield(L, -2, "HTTP_SPECIAL_RESPONSE");
+    lua_setfield(L, -2, "HTTP_MULTIPLE_CHOICES");
```
Comment (agentzh) [inline src/ngx_http_lua_consts.c]: This line breaks compatibility.

File: src/ngx_http_lua_consts.c [hunk 2]
```diff
@@ -89,14 +89,31 @@ ngx_http_lua_inject_http_consts(lua_State *L)
     lua_setfield(L, -2, "HTTP_TRACE");
     /* }}} */
 
+    lua_pushinteger(L, NGX_HTTP_CONTINUE);
+    lua_setfield(L, -2, "HTTP_CONTINUE");
+
+    lua_pushinteger(L, NGX_HTTP_SWITCHING_PROTOCOLS);
+    lua_setfield(L, -2, "HTTP_SWITCHING_PROTOCOLS");
+
+
```
Comment (agentzh) [inline src/ngx_http_lua_consts.c]: One redundant blank line.

File: src/ngx_http_lua_consts.c [hunk 3]
```diff
@@ -112,12 +129,18 @@ ngx_http_lua_inject_http_consts(lua_State *L)
     lua_pushinteger(L, NGX_HTTP_NOT_MODIFIED);
     lua_setfield(L, -2, "HTTP_NOT_MODIFIED");
 
+    lua_pushinteger(L, NGX_HTTP_TEMPORARY_REDIRECT);
+    lua_setfield(L, -2, "HTTP_TEMPORARY_REDIRECT");
+
+
```
Comment (agentzh) [inline src/ngx_http_lua_consts.c]: One redundant blank line.

File: src/ngx_http_lua_consts.c [hunk 4]
```diff
@@ -128,20 +151,50 @@ ngx_http_lua_inject_http_consts(lua_State *L)
...
+    lua_pushinteger(L, 429);
+    lua_setfield(L, -2, "HTTP_TOO_MANY_REQUESTS");
+
+    lua_pushinteger(L, 451);
+    lua_setfield(L, -2, "HTTP_ILLEGAL");
+
+    lua_pushinteger(L, NGX_HTTP_CLOSE);
+    lua_setfield(L, -2, "HTTP_CLOSE");
+
+
```
Comment (agentzh) [inline src/ngx_http_lua_consts.c]: One redundant blank line.

File: src/ngx_http_lua_consts.c [hunk 5]
```diff
@@ -128,20 +151,50 @@ ngx_http_lua_inject_http_consts(lua_State *L)
...
     lua_pushinteger(L, NGX_HTTP_SERVICE_UNAVAILABLE);
     lua_setfield(L, -2, "HTTP_SERVICE_UNAVAILABLE");
 
     lua_pushinteger(L, NGX_HTTP_GATEWAY_TIME_OUT);
     lua_setfield(L, -2, "HTTP_GATEWAY_TIMEOUT");
+
+    lua_pushinteger(L, NGX_HTTP_INSUFFICIENT_STORAGE);
+    lua_setfield(L, -2, "HTTP_INSUFFICIENT_STORAGE");
+
+
```
Comment (agentzh) [inline src/ngx_http_lua_consts.c]: One redundant blank line.

**Comment by msva** [comment]
1)  does redundant blanks interfers something? I added them just for visual niceness. Anyway, I can delete them, if any.

File: src/ngx_http_lua_consts.c (hunk 1 above)
Comment (msva) [inline src/ngx_http_lua_consts.c]: 2) it is. But it is proper name for that. Anyway, I'll fix it, okay.

**Comment by msva** [comment]
@agentzh ping? ;) Can you look into this once again? 😉

**Comment by agentzh** [comment]
@msva Sorry for the delay on my side. Will you update the document in the file `doc/HttpLuaModule.wiki` for this change? Thanks!

**Comment by msva** [comment]
I almost did it already, but should I also add comments like "first added in XXX release"? ;) If so — I need to know, which release will be next: 0.9.16 or rc2 ;)

**Comment by msva** [comment]
anyway, here is next proposition ;)

**Comment by agentzh** [comment]
@msva Cool, thanks!

**Comment by msva** [comment]
there was a little mistake (I confused `HTTP_VERSION_NOT_SUPPORTED`'s code in the source (and wrote fine in wiki). Fixed now ;)

// Btw, I thinking about renaming it to `HTTP_VERSION_UNSUPPORTED` and also about adding CloudFlare's 520 `Unknown Error` ;)

File: doc/HttpLuaModule.wiki [hunk 6]
```diff
@@ -2088,23 +2088,40 @@ These constants are usually used in [[#ngx.location.capture|ngx.location.capture
 '''context:''' ''init_by_lua*, set_by_lua*, rewrite_by_lua*, access_by_lua*, content_by_lua*, header_filter_by_lua*, body_filter_by_lua, log_by_lua*, ngx.timer.*''
 
 <geshi lang="nginx">
+  value = ngx.HTTP_CONTINUE (100) (first added in the v0.9.16 release)
+  value = ngx.HTTP_SWITCHING_PROTOCOLS (102) (first added in the v0.9.16 release)
```
Comment (agentzh) [inline doc/HttpLuaModule.wiki]: This should be 101 instead of 102, right?

File: doc/HttpLuaModule.wiki [hunk 7]
```diff
@@ -2088,23 +2088,40 @@ These constants are usually used in [[#ngx.location.capture|ngx.location.capture
...
   value = ngx.HTTP_UNAUTHORIZED (401)
+  value = ngx.HTTP_PAYMENT_REQUIRED (402) (first added in the v0.9.16 release)
   value = ngx.HTTP_FORBIDDEN (403)
   value = ngx.HTTP_NOT_FOUND (404)
   value = ngx.HTTP_NOT_ALLOWED (405)
+  value = ngx.HTTP_NOT_ACCEPTABLE (406) (first added in the v0.9.16 release)
+  value = ngx.HTTP_REQUEST_TIMEOUT (408) (first added in the v0.9.16 release)
+  value = ngx.HTTP_CONFLICT (409) (first added in the v0.9.16 release)
   value = ngx.HTTP_GONE (410)
+  value = ngx.HTTP_UPGRADE_REQUIRED (426) (first added in the v0.9.16 release)
```
Comment (agentzh) [inline doc/HttpLuaModule.wiki]: You have actually used this 426 status in your real-world setup?

File: doc/HttpLuaModule.wiki [hunk 8]
```diff
@@ -2088,23 +2088,40 @@ These constants are usually used in [[#ngx.location.capture|ngx.location.capture
...
+  value = ngx.HTTP_TOO_MANY_REQUESTS (429) (first added in the v0.9.16 release)
+  value = ngx.HTTP_CLOSE (444) (first added in the v0.9.16 release)
+  value = ngx.HTTP_ILLEGAL (451) (first added in the v0.9.16 release)
```
Comment (agentzh) [inline doc/HttpLuaModule.wiki]: This status code for "Unavailable For Legal Reasons" is not yet universally adopted and at least Microsoft has its own definition, "Redirect".

Comment (msva) [inline doc/HttpLuaModule.wiki]: Yes, but it has some Real-World usage examples, (e.g. livejournal).

File: doc/HttpLuaModule.wiki (hunk 7 above)
Comment (msva) [inline doc/HttpLuaModule.wiki]: Yes, but, actually, not for the browsers, but with custom HTTP-API client :-/

File: doc/HttpLuaModule.wiki (hunk 6 above)
Comment (msva) [inline doc/HttpLuaModule.wiki]: yup. Fixed.

**Comment by msva** [comment]
ping? Should I drop the codes you've commented above, or it is not that necessarily ;)

**Comment by agentzh** [comment]
@msva Patch applied to git master. Thanks for your contribution and sorry for the long delay on my side.

**Pull request closed without merging**